server.py

```diff
@@ -21,16 +21,31 @@ def generate(r):
     for chunk in r.iter_content(chunk_size):
         yield chunk
 
+def is_image(out_path):
+    return out_path.endswith('.jpg') or out_path.endswith('.jpeg') or out_path.endswith('.png') or out_path.endswith('.gif')
+
 def fetch_image(out_path):
     r = requests.get(out_path, stream=True, params=request.args)
     headers = dict(r.headers)
 
+    if headers['Content-Type']  == "binary/octet-stream" and is_image(out_path):
+            headers['Content-Type'] = "image/jpeg"
+    if app.debug:
+        print('serving with headers ' + str(headers))
+
     return Response(generate(r), headers=headers, status=r.status_code)
 
 @app.route('/healthcheck')
 def healthcheck():
     return 'WORKING'
 
+@app.route('/o/<path:uri>')
+def serve_image_replacing_images_mime_type(uri):
```

Comment:
after deploying, the `BACKEND_ASSET_PATH` has to be changed to `d2ssnmyozoo4xu.cloudfront.net` so there is no endless loop